tls/s2n_resume.c

```diff
@@ -155,3 +224,58 @@ int s2n_store_to_cache(struct s2n_connection *conn)
 
     return 0;
 }
+
+int s2n_connection_set_session(struct s2n_connection *conn, const uint8_t *session, size_t length)
+{
+
+    notnull_check(conn);
+    notnull_check(session);
+
+    struct s2n_blob session_data;
+    GUARD(s2n_alloc(&session_data, length));
+    memcpy_check(session_data.data, session, length);
+
+    struct s2n_stuffer from;
+    GUARD(s2n_stuffer_init(&from, &session_data));
+    GUARD(s2n_stuffer_write(&from, &session_data));
+
+    GUARD(s2n_client_deserialize_resumption_state(conn, &from));
+
+    GUARD(s2n_free(&session_data));
```

Comment:
We might memleak here if any of the GUARDs fail before reaching this line.

Reply:
Using goto to do the clean up.